src/smpp_socket.erl

```diff
@@ -174,21 +178,36 @@ send_async(Ref, Pkt) ->
 
 -spec send_async(gen_statem:server_ref(), valid_pdu(), undefined | send_callback()) -> ok.
```

Comment:
```suggestion
-spec send_async(gen_statem:server_ref(), valid_pdu(), undefined | send_callback()) -> ok | {error, overload}.
```

Reply:
Sure, I didn't check the specs in gen_esme carefully. Will look into that a bit later from the laptop)